control-plane/cmd/post-install/kafka_channel_post_migration_deleter.go

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2022 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/system"
+)
+
+type kafkaChannelPostMigrationDeleter struct {
+	k8s kubernetes.Interface
+}
+
+func (d *kafkaChannelPostMigrationDeleter) Delete(ctx context.Context) error {
+	// at this stage,
+	// - we deleted the old controler
+	// - we made the existing channel services to point to new data plane pods
+	// - we can now delete the old dispatcher resources and all the other leftovers
+
+	///////////////////////////////////////////////////////////////////////////////
+	/////////// START DELETING DISPATCHER RESOURCES
+	///////////////////////////////////////////////////////////////////////////////
+
+	// Delete service knative-eventing/kafka-ch-dispatcher
```

Comment:
To be extra safe, I'd check that `kafka-controller` deployment is ready and all KafkaChannels are still ready

Reply:
> all KafkaChannels are still ready
We cannot rely on this because maybe there's a KafkaChannel with a broken spec which cannot be ready because of user error.

But, +1 on checking readiness of `kafka-controller`.